### ISSUE TITLE
Having `java.lang.System` in the `reflect-config.json` results in build error

### ISSUE THREAD

**Issue by collinpeters** [oracle/graal#3447]
**Describe the issue**
We are using `native-image` on a jar tool which scans manifests and code. Recently we added support for `build.gradle` files and groovy was added to our jar. Once that happened it broke our native-image build with the stacktrace at the bottom

**Steps to reproduce the issue**
I have been able to reproduce this with a simple hello world app
1. Create a simple hello world
2. Create a `reflect-config.json` that contains

```
[
{
  "name":"java.lang.System",
  "allDeclaredFields":true,
  "allDeclaredMethods":true,
  "allPublicMethods":true,
  "allDeclaredConstructors":true
}
]
```

3. Compile and build the native image with
```
javac HelloWorld.java

native-image \
  -cp . \
  --verbose \
  --no-server \
  --no-fallback \
  --report-unsupported-elements-at-runtime \
  --enable-http --enable-https \
  --allow-incomplete-classpath \
  -H:ConfigurationFileDirectories=./native-image-configs/ \
  -H:Name=hello-world \
  HelloWorld
```

**Describe GraalVM and your environment:**
 - GraalVM version **21.1**
 - JDK major version: **11**
 - OS: **Linux**
 - Architecture: **AMD64**

**More details**
Consider adding the `--native-image-info` and  `--verbose` flags when building your native image and paste output below.

Full output: https://gist.github.com/collinpeters/871caa586a1f70ef8f83b1e01593a323

Add any other information about the problem here. Especially important are stack traces or log output. Feel free to link to gists or to screenshots if necessary.
```Fatal error:com.oracle.graal.pointsto.util.AnalysisError$ParsingError: Error encountered while parsing jdk.internal.misc.VM.saveAndRemoveProperties(java.util.Properties) 
Parsing context:   
        parsing java.lang.System.initPhase1(System.java:1963)
        parsing com.oracle.svm.reflect.System_initPhase1_5716d11f926794710d77c34fcb247abd3e03878f_34.invoke(Unknown Source)
        parsing java.lang.reflect.Method.invoke(Method.java:566)             
        parsing com.oracle.svm.core.containers.Metrics.systemMetrics(Metrics.java:62)
        parsing com.oracle.svm.core.containers.Container.metrics(Container.java:44)
        parsing com.oracle.svm.core.ContainerInfo.<init>(ContainerInfo.java:34)
        parsing com.oracle.svm.core.Containers.memoryLimitInBytes(Containers.java:162)
        parsing com.oracle.svm.core.heap.PhysicalMemory.doInitialize(PhysicalMemory.java:144)
        parsing com.oracle.svm.core.heap.PhysicalMemory.tryInitialize(PhysicalMemory.java:114)
        parsing com.oracle.svm.core.genscavenge.HeapPolicy.samplePhysicalMemorySize(HeapPolicy.java:351)
        parsing com.oracle.svm.core.genscavenge.ThreadLocalAllocation.runSlowPathHooks(ThreadLocalAllocation.java:161)
        parsing com.oracle.svm.core.genscavenge.ThreadLocalAllocation.slowPathNewArray(ThreadLocalAllocation.java:199)

        at com.oracle.graal.pointsto.util.AnalysisError.parsingError(AnalysisError.java:138)
        at com.oracle.graal.pointsto.flow.MethodTypeFlow.createTypeFlow(MethodTypeFlow.java:327)
        at com.oracle.graal.pointsto.flow.MethodTypeFlow.ensureTypeFlowCreated(MethodTypeFlow.java:302)
        at com.oracle.graal.pointsto.flow.MethodTypeFlow.addContext(MethodTypeFlow.java:103)
        at com.oracle.graal.pointsto.flow.StaticInvokeTypeFlow.update(InvokeTypeFlow.java:420)
        at com.oracle.graal.pointsto.BigBang$2.run(BigBang.java:547)
        at com.oracle.graal.pointsto.util.CompletionExecutor.lambda$execute$0(CompletionExecutor.java:173) 
        at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: org.graalvm.compiler.java.BytecodeParser$BytecodeParserError: com.oracle.svm.core.util.VMError$HostedError: Error in @InjectAccessors handling of field jdk.internal.misc.VM.directMemory, accessors clas
s com.oracle.svm.core.jdk.DirectMemoryAccessors: found no method named set or setDirectMemory
        at parsing jdk.internal.misc.VM.saveAndRemoveProperties(VM.java:207)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.throwParserError(BytecodeParser.java:2601)
        at com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.throwParserError(SharedGraphBuilderPhase.java:106)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.iterateBytecodesForBlock(BytecodeParser.java:3456)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.processBlock(BytecodeParser.java:3258)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.build(BytecodeParser.java:1125)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.buildRootMethod(BytecodeParser.java:1019)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.GraphBuilderPhase$Instance.run(GraphBuilderPhase.java:84)
        at com.oracle.svm.hosted.phases.SharedGraphBuilderPhase.run(SharedGraphBuilderPhase.java:76)
        at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.run(Phase.java:49)
        at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:212)
        at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.apply(Phase.java:42)
        at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.apply(Phase.java:38)
        at com.oracle.graal.pointsto.flow.AnalysisParsedGraph.parseBytecode(AnalysisParsedGraph.java:113)
        at com.oracle.svm.hosted.SVMHost.parseBytecode(SVMHost.java:647)
        at com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsed(AnalysisMethod.java:592)
        at com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.parse(MethodTypeFlowBuilder.java:163)
        at com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.apply(MethodTypeFlowBuilder.java:304)
        at com.oracle.graal.pointsto.flow.MethodTypeFlow.createTypeFlow(MethodTypeFlow.java:313)
        ... 11 more
Caused by: com.oracle.svm.core.util.VMError$HostedError: Error in @InjectAccessors handling of field jdk.internal.misc.VM.directMemory, accessors class com.oracle.svm.core.jdk.DirectMemoryAccessors: found no meth
od named set or setDirectMemory
        at com.oracle.svm.core.util.VMError.shouldNotReachHere(VMError.java:68)
        at com.oracle.svm.hosted.phases.InjectedAccessorsPlugin.error(InjectedAccessorsPlugin.java:152)
        at com.oracle.svm.hosted.phases.InjectedAccessorsPlugin.handleField(InjectedAccessorsPlugin.java:89)
        at com.oracle.svm.hosted.phases.InjectedAccessorsPlugin.handleStoreStaticField(InjectedAccessorsPlugin.java:62)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.genPutStatic(BytecodeParser.java:5060)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.genPutStatic(BytecodeParser.java:5031)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.processBytecode(BytecodeParser.java:5443)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.iterateBytecodesForBlock(BytecodeParser.java:3451)
        ... 26 more
Error: Image build request failed with exit status 1
com.oracle.svm.driver.NativeImage$NativeImageError: Image build request failed with exit status 1
        at com.oracle.svm.driver.NativeImage.showError(NativeImage.java:1772)
        at com.oracle.svm.driver.NativeImage.build(NativeImage.java:1519)
        at com.oracle.svm.driver.NativeImage.performBuild(NativeImage.java:1480)
        at com.oracle.svm.driver.NativeImage.main(NativeImage.java:1467)
        at com.oracle.svm.driver.NativeImage$JDK9Plus.main(NativeImage.java:1959)
```

**Comment by oubidar-Abderrahim** [comment]
Did you try with the latest snapshot build of 21.2.0?

**Comment by collinpeters** [comment]
Just tried with [21.2.0-dev-20210601_2247](https://github.com/graalvm/graalvm-ce-dev-builds/releases/tag/21.2.0-dev-20210601_2247) and got the same build error.

**Comment by collinpeters** [comment]
FWIW If I remove `java.lang.System` from the `reflect-config.json` I am able to build the binary. However when I execute the binary and it gets around to try to invoke some of this groovy related code I get this error:
```
Exception in thread "main" com.oracle.svm.core.jdk.UnsupportedFeatureError: Defining classes from new bytecodes run time.
	at com.oracle.svm.core.util.VMError.unsupportedFeature(VMError.java:87)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:425)
	at org.codehaus.groovy.reflection.ClassLoaderForClassArtifacts.define(ClassLoaderForClassArtifacts.java:43)
	at org.codehaus.groovy.reflection.ClassLoaderForClassArtifacts$1.run(ClassLoaderForClassArtifacts.java:76)
	at org.codehaus.groovy.reflection.ClassLoaderForClassArtifacts$1.run(ClassLoaderForClassArtifacts.java:74)
	at java.security.AccessController.doPrivileged(AccessController.java:82)
	at org.codehaus.groovy.reflection.ClassLoaderForClassArtifacts.defineClassAndGetConstructor(ClassLoaderForClassArtifacts.java:74)
	at org.codehaus.groovy.runtime.callsite.CallSiteGenerator.compilePojoMethod(CallSiteGenerator.java:250)
	at org.codehaus.groovy.reflection.CachedMethod.createPojoMetaMethodSite(CachedMethod.java:267)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.createCachedMethodSite(PojoMetaMethodSite.java:162)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.createPojoMetaMethodSite(PojoMetaMethodSite.java:151)
	at groovy.lang.MetaClassImpl.createPojoCallSite(MetaClassImpl.java:3408)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.createPojoSite(CallSiteArray.java:131)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.createCallSite(CallSiteArray.java:165)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:120)
	at org.codehaus.groovy.ast.builder.AstBuilder.buildFromString(AstBuilder.groovy:102)
	at org.codehaus.groovy.ast.builder.AstBuilder.buildFromString(AstBuilder.groovy)

```

**Comment by collinpeters** [comment]
I'm not sure if it is related yet but I also have to `-H:ServiceLoaderFeatureExcludeServices=java.nio.charset.spi.CharsetProvider` once this groovy stuff is loaded. That service contains a single line to a truezip class `de.schlichtherle.truezip.nio.charset.ZipCharsetProvider` that complains about being loaded at build time. This error doesn't happen if groovy is excluded.

**Comment by oubidar-Abderrahim** [comment]
Tracked internally at GR-31871

**Comment by zapster** [comment]
Hi @collinpeters!

The issue is already fixed in the repos. So starting with [21.3.0](https://www.graalvm.org/release-notes/version-roadmap/), the above example should "work". I'm using quotes here because you will still not be able to reflectively access all declared methods from `java.lang.System`, simply because some of them are not supported by native image. However, it will not be reported at image generation time but at run time (because of `--report-unsupported-elements-at-runtime`).

So long story short, having the following...
```json
[
{
  "name":"java.lang.System",
  "allDeclaredMethods":true,
}
]
```
... in the reflection config is not really useful because it will fail, either when accessing it at run time (`--report-unsupported-elements-at-runtime`) or already at native image generation time.
